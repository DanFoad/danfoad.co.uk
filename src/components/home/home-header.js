import React from 'react';

import AppNav from '../app/app-nav';
import HomeHeaderPipeline from './home-header-pipeline';

export default class HomeHeader extends React.Component {
  getHeader() {
    return (
      <div className="home-header__contents">
        <h1 className="home-header__name">Dan Foad</h1>
        <div className="home-header__title">
          <h3>
            <img className="home-header__logo" src="/skylogo.png" alt="" />
            Associate Software Developer
          </h3>
          <p>Currently working with NowTV</p>
        </div>
        <div className="home-header__buttons">
          <button className="home-header__contact">
            <img src="/contact.png" alt="" />
            <span>Contact Me</span>
          </button>
        </div>
      </div>
    );
  }

  render() {
    return (
      <header className="home-header">
        <AppNav />
        {this.getHeader()}
        <HomeHeaderPipeline />
      </header>
    );
  }
}

HomeHeader.propTypes = {};
