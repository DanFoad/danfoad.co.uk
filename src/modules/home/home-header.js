import React from 'react';

import { t } from '../../services/label-service';

import { HomeHeaderPipeline } from './home-header-pipeline';

export class HomeHeader extends React.Component {
  render() {
    return (
      <header className="home-header">
        <div className="home-header__contents">
          <div className="home-header__name">
            <h1>{t('general.name')}</h1>
          </div>
          <div className="home-header__title">
            <h3>
              <img
                className="home-header__logo"
                src="/img/skylogo.png"
                alt=""
              />
              {t('general.role')}
            </h3>
            <p>{t('home.header.subline')}</p>
          </div>
          <div className="home-header__buttons">
            <button className="home-header__contact" type="button">
              <img src="/img/contact.png" alt="" />
              <span>{t('home.header.contact')}</span>
            </button>
          </div>
        </div>
        <HomeHeaderPipeline />
      </header>
    );
  }
}

HomeHeader.propTypes = {};
