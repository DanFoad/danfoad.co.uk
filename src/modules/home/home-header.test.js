import React from 'react';
import { shallow } from 'enzyme';

import { HomeHeader } from './home-header';

describe('home-header', () => {
  it('should match snapshot', () => {
    expect(shallow(<HomeHeader />)).toMatchSnapshot();
  });
});
