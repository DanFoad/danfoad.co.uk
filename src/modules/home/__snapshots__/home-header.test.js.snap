// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-header should match snapshot 1`] = `
<header
  className="home-header"
>
  <div
    className="home-header__contents"
  >
    <div
      className="home-header__name"
    >
      <h1>
        translated general.name
      </h1>
    </div>
    <div
      className="home-header__title"
    >
      <h3>
        <img
          alt=""
          className="home-header__logo"
          src="/img/skylogo.png"
        />
        translated general.role
      </h3>
      <p>
        translated home.header.subline
      </p>
    </div>
    <div
      className="home-header__buttons"
    >
      <button
        className="home-header__contact"
        type="button"
      >
        <img
          alt=""
          src="/img/contact.png"
        />
        <span>
          translated home.header.contact
        </span>
      </button>
    </div>
  </div>
  <Connect(HomeHeaderPipeline) />
</header>
`;
